File: Dec2020/27Dec/AvailableCapturesForRook.cpp
/*
Problem name: 999. Available Captures for Rook
Problem link: https://leetcode.com/problems/available-captures-for-rook/
*/

class Solution {
public:
    int numRookCaptures(vector<vector<char>>& board) {
        int x, y, captured=0;
        for(int i=0;i<8;++i){
            for(int j=0;j<8;++j ){
                if(board[i][j] == 'R'){
                    x=i;
                    y=j;
                    break;
                }
            }
        }
        
       for(int i=x-1;i>=0;--i){     //top
            if(board[i][y] == 'p'){
                captured++;
                break;
            }
            else if(board[i][y] == 'B')
                break; 
        } 
        
        for(int i=x+1;i<8;++i){      //bottom
            if(board[i][y] == 'p'){
                captured++;
                break;
            }
            else if(board[i][y] == 'B')
                break;
        }
        
        for(int i=y-1;i>=0;--i){        //left
            if(board[x][i] == 'p'){
                captured++;
                break;
            }
            else if(board[x][i] == 'B')
                break;
        } 
        
        for(int i=y+1;i<8;++i){      //right
            if(board[x][i] == 'p'){
                captured++;
                break;
            }
            else if(board[x][i] == 'B')
                break;
        }
        
        return captured;
    }
};

File: Dec2020/27Dec/AvailableCapturesForRook.java
/*
Problem Name: 999. Available Captures for Rook
Problem Link: https://leetcode.com/problems/available-captures-for-rook/
*/
class Solution {
    public int numRookCaptures(char[][] board) {
        int rr = -1, rc = -1;
        for(int i=0;i<8;i++){
            for(int j=0;j<8;j++){
                if(board[i][j] == 'R'){
                    rr = i;
                    rc = j;
                    break;
                }
            }
            if(rr != -1)
                break;
        }
        int pawns = 0;
        for(int i=rc-1;i>=0;i--){
            if(board[rr][i] == 'p'){ pawns++; break; }
            else if(board[rr][i] == 'B'){ break; }
        }
        for(int i=rr-1;i>=0;i--){
            if(board[i][rc] == 'p'){ pawns++;break; }
            else if(board[i][rc] == 'B'){ break; }
        }
        for(int i=rc+1;i<8;i++){
            if(board[rr][i] == 'p'){ pawns++; break; }
            else if(board[rr][i] == 'B'){ break; }
        }
        for(int i=rr+1;i<8;i++){
            if(board[i][rc] == 'p'){ pawns++;break; }
            else if(board[i][rc] == 'B'){ break; }
        }
        return pawns;
    }
}
